### PR TITLE
Add PrepareAndSignTransaction example

### PR DESCRIPTION
**Pull request by samuel-rufi**

### CODE REVIEW

File: bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/PrepareAndSignTransaction.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import java.util.Map;
+
+public class PrepareAndSignTransaction {
+    public static void main(String[] args) throws ClientException {
+        // Build the client.
+        Client client = new Client(new ClientConfig().withNodes(new String[]{"https://api.testnet.shimmer.network"}));
+
+        // Build the secret manager.
+        SecretManager secretManager = new MnemonicSecretManager("YOUR_SECRET_MNEMONIC");
```
Comment (Thoralf-M) [inline bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/PrepareAndSignTransaction.java:19]: Maybe have at least a comment that the user needs to replace `YOUR_SECRET_MNEMONIC`? Might be confusing if someone isn't familiar with what a mnemonic is

Comment (samuel-rufi) [reply bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/PrepareAndSignTransaction.java:19]: good point

Comment (samuel-rufi) [reply bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/PrepareAndSignTransaction.java:19]: fixed

**Review by Thoralf-M — APPROVED** [review]

File: bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/PrepareAndSignTransaction.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        // Build the secret manager.
+        // NOTE: `YOUR_SECRET_MNEMONIC` serves  as a placeholder. Replace it with the mnemonic you want to use to sign the transaction.
+        SecretManager secretManager = new MnemonicSecretManager("YOUR_SECRET_MNEMONIC");
+
+        // Build the output for the transaction.
+        String receiverAddress = client.generateAddresses(secretManager, new GenerateAddressesOptions().withRange(new Range(0, 1)))[0];
+        BuildBlockOptions.ClientBlockBuilderOutputAddress output = new BuildBlockOptions.ClientBlockBuilderOutputAddress(receiverAddress, Integer.toString(1000000));
+
+        // Build block.
+        Map.Entry<BlockId, Block> b = client.buildAndPostBlock(secretManager, new BuildBlockOptions());
```
Comment (thibault-martinez) [inline bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/PrepareAndSignTransaction.java:26]: Just curious, can this return many blocks or are there no tuples in Java?

Comment (samuel-rufi) [reply bindings/java/iota-client-java/examples/ExampleProject/src/node_api_core/PrepareAndSignTransaction.java:26]: Nope, there are no tuples in Java.
This function returns a single Key-Value pair, so basically an BlockId -> Block entry

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**